tcks/microprofile-rest-client/src/test/java/io/quarkus/tck/restclient/SkipTestsThatThrowNoClassDefFound.java

```diff
@@ -0,0 +1,44 @@
+package io.quarkus.tck.restclient;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.microprofile.rest.client.tck.ClientHeaderParamTest;
+import org.eclipse.microprofile.rest.client.tck.InvokeWithJsonBProviderTest;
+import org.eclipse.microprofile.rest.client.tck.InvokeWithJsonPProviderTest;
+import org.testng.IAnnotationTransformer;
+import org.testng.IConfigurable;
+import org.testng.IConfigureCallBack;
+import org.testng.ITestNGMethod;
+import org.testng.ITestResult;
+import org.testng.annotations.ITestAnnotation;
+
+public class SkipTestsThatThrowNoClassDefFound implements IConfigurable, IAnnotationTransformer {
```

Comment:
What's this for? Shouldn't we just exclude those tests?

Reply:
I tried, but unfortately you can't just exclude classes in the testsuite (or at least I didn't find a way to do that), since exclusion is based on methods. This means that the `@BeforeTest` stuff still runs and causes an exception even when all test methods are excluded from a class.
So this class is a workaround to ensure that these misbehaving tests won't fail the testsuite.

Reply:
Hm, you could try to use `<exclude name=".*"/>` - this should exclude all test methods and I would expect the TestNG to completely ignore the test class.

Reply:
I tried again the following:

```
            <class name="org.eclipse.microprofile.rest.client.tck.ClientHeaderParamTest">
                <methods>
                    <exclude name=".*" />
                </methods>
            </class>
            <class name="org.eclipse.microprofile.rest.client.tck.InvokeWithJsonBProviderTest">
                <methods>
                    <exclude name=".*" />
                </methods>
            </class>
            <class name="org.eclipse.microprofile.rest.client.tck.InvokeWithJsonPProviderTest">
                <methods>
                    <exclude name=".*" />
                </methods>
            </class>
```

However the problem (of the `@BeforeTest` methods being called) still persists

Reply:
Ok.